tools/python/src/face_recognition.cpp

```diff
@@ -47,33 +48,93 @@ class face_recognition_model_v1
         const int num_jitters
     )
     {
+        std::vector<numpy_image<rgb_pixel>> batch_img(1, img);
+        std::vector<std::vector<full_object_detection>> batch_faces(1, faces);
+        return batch_compute_face_descriptors(batch_img, batch_faces, num_jitters)[0];
+    }
+
+    std::vector<std::vector<matrix<double,0,1>>> batch_compute_face_descriptors (
+        const std::vector<numpy_image<rgb_pixel>>& batch_imgs,
+        const std::vector<std::vector<full_object_detection>>& batch_faces,
+        const int num_jitters
+    )
+    {
+
+        if (batch_imgs.size() != batch_faces.size())
+            throw dlib::error("The array of images and the array of array of locations must be of the same size");
 
-        for (auto& f : faces)
+        int total_chips = 0;
+        for (auto& faces : batch_faces)
         {
-            if (f.num_parts() != 68 && f.num_parts() != 5)
-                throw dlib::error("The full_object_detection must use the iBUG 300W 68 point face landmark style or dlib's 5 point style.");
+            total_chips += faces.size();
+            for (auto& f : faces)
+            {
+                if (f.num_parts() != 68 && f.num_parts() != 5)
+                    throw dlib::error("The full_object_detection must use the iBUG 300W 68 point face landmark style or dlib's 5 point style.");
+            }
         }
 
 
-        std::vector<chip_details> dets;
-        for (auto& f : faces)
-            dets.push_back(get_face_chip_details(f, 150, 0.25));
         dlib::array<matrix<rgb_pixel>> face_chips;
-        extract_image_chips(img, dets, face_chips);
+        for (int i = 0; i < batch_imgs.size(); ++i)
+        {
+            auto& faces = batch_faces[i];
+            auto& img = batch_imgs[i];
+
+            std::vector<chip_details> dets;
+            for (auto& f : faces)
+                dets.push_back(get_face_chip_details(f, 150, 0.25));
+            dlib::array<matrix<rgb_pixel>> this_img_face_chips;
+            extract_image_chips(img, dets, this_img_face_chips);
 
-        std::vector<matrix<double,0,1>> face_descriptors;
-        face_descriptors.reserve(face_chips.size());
+            for (auto& chip : this_img_face_chips)
+                face_chips.push_back(chip);
+        }
 
+        std::vector<std::vector<matrix<double,0,1>>> face_descriptors(batch_imgs.size());
         if (num_jitters <= 1)
         {
+            int img_i = 0;
+            int i = 0;
             // extract descriptors and convert from float vectors to double vectors
             for (auto& d : net(face_chips,16))
-                face_descriptors.push_back(matrix_cast<double>(d));
+            {
```

Comment:
Sorry, but these loops are still freaking me out.  I think you should be able to use this instead:

```
auto descriptors = net(face_chips,16);
auto next = descriptors.begin();
for (auto& img : face_descriptors)
{
    for (auto& face : img)
        face = matrix_cast<double>(*next++);
}
```

Does that work or am I missing something?  All you need to do is initialize face_descriptors appropriately.

Reply:
I'm sorry, I don't understand your example. What is to be done with `face` then?